source/code/source/playfab/PlayFabEventPipeline.cpp

```diff
@@ -159,15 +158,26 @@ namespace PlayFab
         using clock = std::chrono::steady_clock;
         using Result = PlayFabEventBuffer::EventConsumingResult;
         std::shared_ptr<const IPlayFabEmitEventRequest> request;
-        size_t batchCounter = 1; // used to track uniqueness of batches in the map
+        uintptr_t batchCounter = 1; // used to track uniqueness of batches in the map
         std::chrono::steady_clock::time_point momentBatchStarted; // used to track when a currently assembled batch got its first event
+        
+        std::vector<std::shared_ptr<const IPlayFabEmitEventRequest>> batch;
+
+        batch.reserve(this->settings->maximalNumberOfItemsInBatch);
```

Comment:
this reserve statement can throw. I think that means you want to hoist the `try {}` block so it wraps the entire while loop. You've got an exception block which seems to allow continuing the while loop even if an exception fires (see line 246) but that doesn't seem intentional so just hoisting the try block seems like the right fix